vcf/reader.py

```diff
@@ -0,0 +1,48 @@
+import itertools
+
+
+class MultiVCFReader:
```

Comment:
With current implementation, this class does more than reading multiple VCFs. It is aware of the "merging" operation being performed on the VCFs. If we continue with current implementation, I would suggest to change the class name to something more indicative.

Ideally, I would suggest not to mix the logics of reading a VCF with how the stuff just read from the VCF is going to be used, based on the Single Responsibility Principle (SRP). You don't want to modify the same class if you want to modify what is retrieved from the VCF and if you want to merge the retrieved contents in different way.